src/core/api/border_agent_api.cpp

```diff
@@ -42,6 +42,13 @@
 
 using namespace ot;
 
+#if OPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE
```

Comment:
Need to define a default value.

Reply:
Added to `config/border_agent.h` in the new push